lsp-mode.el

```diff
@@ -1631,10 +1631,10 @@ On other systems, returns path without change."
                           uri)
                       ;; `url-generic-parse-url' is buggy on windows:
                       ;; https://github.com/emacs-lsp/lsp-mode/pull/265
-                      (or (and (eq system-type 'windows-nt)
-                               (eq (elt file 0) ?\/)
-                               (substring file 1))
-                          file))))
+                      (if (and (eq system-type 'windows-nt)
+                               (s-prefix? "/c:" file))
```

Comment:
This is not really correct, since the path can start from other drive (`/d:`, `/e:` ...), so you probably need to pattern matching it.
What wrong with the original code?

Reply:
https://github.com/emacs-lsp/lsp-mode/blob/1805c9caa146f60a5d9109f2a14d3596a5d1228f/test/lsp-common-test.el#L38 and https://github.com/emacs-lsp/lsp-mode/blob/1805c9caa146f60a5d9109f2a14d3596a5d1228f/test/lsp-common-test.el#L54 failed on Windows before making this change, because the leading forward slash in something like "/root/usr/" got trimmed off.

Reply:
I see.
I think we should also fix the test instead of production code. The `"/root/%5E/%60"` is not suppose to be a path in Windows.
Also in https://github.com/emacs-lsp/lsp-mode/pull/265, the reason the first slash is removed in Windows is because of `file:///` which is prefix in Windows will be parse as only `file://` and left the final `/` in the path by `url-generic-parse-url`.
The original check would be enough

Reply:
OK, I went back to the original check and used `'gnu-linux` as `system-type` in the offending tests.